src/main/java/ch/uzh/ifi/fs22/sel/group15/cozycave/server/service/ApplicationService.java

```diff
@@ -0,0 +1,182 @@
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service @Transactional public class ApplicationService {
+
+    private final Logger log = LoggerFactory.getLogger(ApplicationService.class);
+
+    private final ApplicationRepository applicationRepository;
+    private final ListingRepository listingRepository;
+    private final UserRepository userRepository;
+
+    @Autowired public ApplicationService(@Qualifier("applicationRepository") ApplicationRepository applicationRepository, ListingRepository listingRepository, UserRepository userRepository) {
+        this.applicationRepository = applicationRepository;
+        this.listingRepository = listingRepository;
+        this.userRepository = userRepository;
+    }
+
+    // get all Applications of a user > id = user id
+    public List<Application> findApplicationsOfUser(UUID id) {
+        return this.applicationRepository.findByApplicant_Id(id);
+    }
+    // get all Applications of a user > id = user id
+    public List<Application> findApplicationsOfUser(User user) {
+        return this.applicationRepository.findByApplicant_Id(user.getId());
+    }
+
+    // get all Applications for a listing > id = listing id
+    public List<Application> findApplicationsToListing(UUID id) {
+        return this.applicationRepository.findByListing_Id(id);
+    }
+
+    // get all Applications for a listing > id = listing id
+    public List<Application> findApplicationsToListing(Listing listing) {
+        return this.applicationRepository.findByListing_Id(listing.getId());
+    }
+
+    // get concrete application > id = application id
+    public Optional<Application> findApplicationById(UUID id) {
+        return this.applicationRepository.findById(id);
+    }
+    // get concrete application > id = application id
+    public Optional<Application> findApplicationById(Application application) {
+        return this.applicationRepository.findById(application.getId());
+    }
+
+    public @NotNull Application createApplication(Application newApplication) {
+        log.debug("create Application {}", newApplication);
+        checkIfDataIsValid(newApplication);
+
+        newApplication.setId(UUID.randomUUID());
+        newApplication.setCreationDate(new Date());
+        // manually set application to pending regardless of how application was sent
+        newApplication.setApplication_status(ApplicationStatus.PENDING);
+        newApplication.setId(UUID.randomUUID());
+
+        Listing listing = listingRepository.getOne(newApplication.getListing().getId());
```

Comment:
`.getOne(...)` can throw an `javax.persistence.EntityNotFoundException` which is not handled. either handle the exception and throw a proper `ReponseStatusException` or use the findBy and check on `Optional<T>#isPresent`

Reply:
same in other parts of the code. I didn't mark them